### PR TITLE
Add RPM Repository tests

### PR DESCRIPTION
**Pull request by Ichimonji10**
Add RPM repository tests. Module `pulp_smash.tests.rpm.api_v2.test_repo_create`
includes the following test cases:

* `CreateTestCase` ensures it is possible to create repositories with and
  without feeds.
* `SyncValidFeedTestCase` and `SyncInvalidFeedTestCase` ensure that Pulp
  correctly handles sync requests, regardless of whether the feed is valid.
* `PublishTestCase` ensures it is possible to upload an RPM file to a
  repository, copy the repository's contents to a second repository, add a
  distributor to the first repository, publish the first repository, and
  download the original RPM.

Test results:

    $ python -m unittest2 pulp_smash.tests.rpm.api_v2.test_sync_publish
    .............s........
    ----------------------------------------------------------------------
    Ran 22 tests in 27.118s

    OK (skipped=1)

Many thanks to @dkliban for help with designing these tests.

### CODE REVIEW

**Comment by Ichimonji10** [comment]
This PR isn't quite ready for a merge yet. But I'd like to get some early review from @dkliban, or anyone else with expertise with RPM repositories. I've reworked the exact form of these tests pretty heavily from when they were first prototyped, and it'd be good to know if I broke anything. Also, the `test_error_details` test case is failing, and I think there was a bug for that. Can anyone point me to the right bug in our bug tracker?

File: pulp_smash/tests/rpm/api_v2/test_repo_create.py [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+                self.assertEqual(body['importer_' + key], importers[0][key])
+
+    @classmethod
+    def tearDownClass(cls):
+        """Destroy the created RPM repositories."""
+        for attrs in cls.attrs_iter:
+            _delete(cls.cfg, attrs['_href'])
+
+
+class SyncFeedTestCase(TestCase):
```
Comment (dkliban) [inline pulp_smash/tests/rpm/api_v2/test_repo_create.py]: Should this be SyncInvalidFeedTestCase?

File: pulp_smash/tests/rpm/api_v2/test_repo_create.py [hunk 2]
```diff
@@ -0,0 +1,662 @@
...
+        self.assertIn('spawned_tasks', self.response.json())
+
+    def test_spawned_tasks_v2(self):
+        """The server should return exactly one spawned task."""
+        task_bodies = tuple(chain.from_iterable(
+            _poll_tasks(self.cfg, spawned_task['_href'])
+            for spawned_task in self.response.json()['spawned_tasks']
+        ))
+        self.assertEqual(len(task_bodies), 1, task_bodies)
+
```
Comment (dkliban) [inline pulp_smash/tests/rpm/api_v2/test_repo_create.py]: There should probably be a test to check that an error was reported.

File: pulp_smash/tests/rpm/api_v2/test_repo_create.py (hunk 1 above)
Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_repo_create.py]: The `SyncFeedTestCase` and `SyncInvalidFeedTestCase` have basically identical set-up procedures. I might as well just combine the two classes. :+1:

Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_repo_create.py]: In fact, I'm not sure I really even need any methods from this class except `test_status_code`. Calling `poll_tasks()` implicitly ensures that tasks have been spawned.

File: pulp_smash/tests/rpm/api_v2/test_sync_publish.py [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+        'importer_type_id': 'yum_importer',
+        'notes': {'_repo-type': 'rpm-repo'},
+    }
+
+
+class _TaskTimedOutException(Exception):
+    """Indicates that polling a task timed out."""
+
+
+def _poll_tasks(server_config, href):
```
Comment (dkliban) [inline pulp_smash/tests/rpm/api_v2/test_sync_publish.py]: Would it be better to put this in some place that is shared by other tests? We need to poll tasks for other plugins also.

File: pulp_smash/tests/rpm/api_v2/test_sync_publish.py [hunk 4]
```diff
@@ -0,0 +1,650 @@
...
+
+    """
+    return _handle_response(requests.post(
+        server_config.base_url + href + 'actions/associate/',
+        json={'source_repo_id': source_repo_id},
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _add_yum_distributor(server_config, href, responses=None):
```
Comment (dkliban) [inline pulp_smash/tests/rpm/api_v2/test_sync_publish.py]: Could this be more generic? We need to have ability to add any type of supported distributor. Other tests will add different types of distributors.

File: pulp_smash/tests/rpm/api_v2/test_sync_publish.py (hunk 3 above)
Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_sync_publish.py]: Yes. But I'd like to do that in a separate commit.

Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_sync_publish.py]: Several of the other helper functions can also be extracted from this module too.

File: pulp_smash/tests/rpm/api_v2/test_sync_publish.py (hunk 4 above)
Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_sync_publish.py]: I don't know how easy or hard that is to accomplish. What differs between the payload for creating a yum distributor and the payload for creating the other types of distributors? Just the `distributor_type_id`?

**Comment by dkliban** [comment]
I started putting comments on everything that I thought could live outside the test. I think it would be perfectly OK to move those methods in a separate commit. 

This looks goo to merge.

File: pulp_smash/tests/rpm/api_v2/test_sync_publish.py (hunk 4 above)
Comment (dkliban) [inline pulp_smash/tests/rpm/api_v2/test_sync_publish.py]: I believe that the only difference is the distributor_type_id. I will confirm this and this can be updated in a separate PR.

Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_sync_publish.py]: Cool. If it's just the `distributor_type_id`, then something like this could be written:

```python
def add_distributor(server_config, href, distributor_type_id, responses=None):
    pass
```

**Comment by Ichimonji10** [comment]
Thank you @dkliban. See #40.
